numpy/core/src/multiarray/dtype_transfer.c

```diff
@@ -3482,8 +3521,13 @@ PyArray_GetDTypeCopySwapFn(int aligned,
                                     itemsize);
         *outtransferdata = NULL;
     }
+    else if (dtype->kind == 'U') {
+        return wrap_copy_swap_function(aligned,
+                                       src_stride, dst_stride, dtype, 1,
+                                       outstransfer, outtransferdata);
```

Comment:
Might this be changed to a call to `PyArray_GetStridedZeroPadCopyFn`? I would imagine that would be a small performance gain over using the element-wise `copy_swap` function.